src/setDay/index.js

```diff
@@ -25,14 +25,14 @@ import requiredArgs from '../_lib/requiredArgs/index.js'
  * @throws {RangeError} `options.weekStartsOn` must be between 0 and 6
  *
  * @example
- * // Set Sunday to 1 September 2014:
- * var result = setDay(new Date(2014, 8, 1), 0)
+ * // Set week day to the 0th day, with the default weekStartsOn of Sunday:
+ * var result = setDay(new Date('2014-09-01') /* a Monday */, 0)
  * //=> Sun Aug 31 2014 00:00:00
  *
  * @example
- * // If week starts with Monday, set Sunday to 1 September 2014:
- * var result = setDay(new Date(2014, 8, 1), 0, { weekStartsOn: 1 })
- * //=> Sun Sep 07 2014 00:00:00
+ * // Set week day to the 0th day, with a weekStartsOn of Monday:
+ * var result = setDay(new Date('2014-09-01'), 0, { weekStartsOn: 1 })
+ * //=> Monday Sep 01 2014 00:00:00
```

Comment:
@BoomDev it'd probably be a good idea to add this as a test too

Reply:
@dcousens sounds great, will do.

One comment - and maybe this is being picky but please advise - the great majority of 'new Date()'
used in the code including the documentation is using this format:

new Date(2014, 8, 1)

Instead of your proposed:

new Date('2014-09-01')

...which I like better because it's more readable but wouldn't we want to be consistent with the rest?

Reply:
wait... these test are already there!

@dcousens I think what you suggest is incorrect.

In setDay(date, **day**, [options]), my understanding was that 'day' is the day of the week of the new date (as in 0=Sun...), not the index of the day of the new date relative to the weekStartsOn.

In other words :
setDay(new Date('2014-09-01'), 0 /*a Sunday*/, { weekStartsOn: 1 })
=> **Sun** Sep 07 2014

And _not_:
setDay(new Date('2014-09-01'), 0 /*index0 of week starting on a Monday*/, { weekStartsOn: 1 /* Monday */})

Reply:
> ...which I like better because it's more readable but wouldn't we want to be consistent with the rest?

Indeed!  I couldn't help but question why the examples used that format for `new Date`.

Reply:
I think I need to review this function before I comment further,  the functionality isn't appearing straight forward to me.

Reply:
@dcousens sure. The way I see it is, this is especially useful for locales where the week starts on Mondays (other weekStartsOn are edge cases I guess - as probably are the cases where day <0 or >6 , but OK).
So for example if you want to get the Sunday of a date's week, if you consider that that week starts on Mondays and your date is say a Tuesday, what you're looking for is the Sunday after that Tuesday, not the one before.
In contrast, what you suggested, taking the same example, was resulting on the Monday before that Tuesday. Which in fact you could get at by simply doing this:
setDay(new Date(2014, 9, 2 /* a Tuesday */), 1) 

Hope this helps.

Reply:
@BoomDev thank you!  
Your phrasing
> if you want to get the Sunday of a date's week, 

Clicked it for me.    Thank you!   The `weekStartsOn` variable is for resolving the ambiguity of whether to use the Sunday from the before or after the given date.

Reply:
@dcousens you're most welcome, happy it helped!

Reply:
@dcousens what are the next steps ? Anything I can do to complete the process? Sorry for my ignorance, it's [shame!] my first contribution to open source...

Reply:
@BoomDev nothing to do here!  You PR is on the other repository and it is waiting for review from a contributor.
I had simply hoped to contribute what I could here,  but,  as you thankfully pointed out my,  my understanding was flawed.

:+1:

Reply:
@dcousens thanks for your help! Cheers!

Reply:
PS.  Congratulations on contributing back!

Reply:
> PS. Congratulations on contributing back to OSS!

Thanks! I'm glad I did. The code I wrote to fix the bug can probably be nicer but I though better contribute with working but ugly code than do nothing :-D